File: September2021/OrderlyQueue.java
class Solution {
    public String orderlyQueue(String s, int k) {
        if(k==0){
            return s;
        }
        else if (k > 1) {
            char[] charS = s.toCharArray();
            Arrays.sort(charS);
            return new String(charS);

        } else if(k==1) {
            String ans = s;
            for (int i = 1; i < s.length(); i++) {
                String reorderedStr = s.substring(i) + s.substring(0, i);
                if (ans.compareTo(reorderedStr) > 0) {
                    ans = reorderedStr;
                }
            }
            return ans;
        }
        return s;

    }
}

File: November2022/C++ Sol/orderly-queue.cpp
// @Sarang661
class Solution {
public:
    string orderlyQueue(string s, int k) {
        if(k==0){
            return s;
        }
        string ans = s;
        int n = s.size();
        if(k==1){
            for(int i=0;i<s.size();i++){
                 string curr = s.substr(i+1,n-(i+1)) + s.substr(0,i+1);
                 ans = min(ans,curr);
            }
            return ans;
        }
        else{
            sort(s.begin(),s.end());
            return s;
        }
    }
};